tests/basic.rs

```diff
@@ -0,0 +1,220 @@
+
+mod mocks;
+
+use mocks::dtls::{self, Config};
+use mocks::transport;
+use tokio;
+use tokio::time::{sleep, Duration};
+use std::sync::{Arc, Mutex};
+
+const TEST_MESSAGE: &str = "Hello world";
+const TEST_TIME_LIMIT: Duration = Duration::from_secs(5);
+const MESSAGE_RETRY: Duration = Duration::from_millis(200);
+
+pub async fn random_port() -> u16 {
+    let addr = "127.0.0.1:0".parse::<std::net::SocketAddr>().unwrap();
+    let sock = match tokio::net::UdpSocket::bind(addr).await {
+        Ok(s) => s,
+        Err(e) => panic!(e),
+    };
+    let local_addr: std::net::SocketAddr = match sock.local_addr() {
+        Ok(s) => s,
+        Err(e) => panic!(e),
+    };
+    local_addr.port()
+}
+
+// Spawn and await tasks to read from and write to the given stream
+pub async fn simple_read_write<'a>(
+    stream: tokio::net::TcpStream,
+    out_buffer: Arc<Mutex<[u8; 8192]>>,
+    writer_join_handle: &mut tokio::task::JoinHandle<Result<(), std::io::Error>>,
+    reader_join_handle: &mut tokio::task::JoinHandle<Result<(), std::io::Error>>,
+) -> Result<(), std::io::Error>
+{
+    let ref arc_mx_stream = Arc::new(Mutex::new(stream));
+    let read = Arc::clone(arc_mx_stream);
+    // Read from stream into out buffer
+    *reader_join_handle = tokio::spawn( async {
```

Comment:
It seem like I'm not using `Arc<Mutex<_>>` properly here. Is there another way to pass the buffer back up the execution chain?

```
future cannot be sent between threads safely

future created by async block is not `Send`

help: within `impl std::future::Future`, the trait `std::marker::Send` is not implemented for `std::sync::MutexGuard<'_, tokio::net::TcpStream>`
```

Reply:
Has this been fixed? Seems outdated.

Reply:
Yes, it was primarily because of not using tokio's `Mutex` implementation and instead using the one in `std`.